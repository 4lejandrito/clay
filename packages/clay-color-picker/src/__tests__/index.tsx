/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ClayColorPicker from '..';
import ReactDOM from 'react-dom';
import {act, Simulate} from 'react-dom/test-utils';

describe('ClayColorPicker', () => {
	let container: HTMLElement | null;

	beforeEach(() => {
		container = document.createElement('div');
		document.body.appendChild(container);
	});

	afterEach(() => {
		if (container) {
			document.body.removeChild(container);
			container = null;
		}
	});

	it('renders', () => {
		act(() => {
			ReactDOM.render(
				<ClayColorPicker
					label="Default Colors"
					name="colorPicker1"
					onValueChange={() => {}}
					spritemap="/test/path"
					title="Default"
					value="FFF"
				/>,
				container
			);
		});

		expect(container).toMatchSnapshot();
	});

	it('renders with menu displayed', () => {
		act(() => {
			ReactDOM.render(
				<ClayColorPicker
					label="Default Colors"
					name="colorPicker1"
					onValueChange={() => {}}
					spritemap="/test/path"
					title="Default"
					value="FFF"
				/>,
				container
			);
		});

		const dropdownToggle = (container as HTMLElement).querySelector(
			'.dropdown-toggle'
		);

		act(() => {
			if (dropdownToggle) {
				Simulate.click(dropdownToggle);
			}
		});

		expect(container).toMatchSnapshot();
	});

	it('displays custom colors', () => {
		act(() => {
			ReactDOM.render(
				<ClayColorPicker
					colors={['008000', '00FFFF', '0000FF']}
					label="Custom Colors"
					name="colorPicker2"
					onColorsChange={() => {}}
					onValueChange={() => {}}
					spritemap="/test/path"
					title="Custom Colors"
					value="008000"
				/>,
				container
			);
		});

		const dropdownToggle = (container as HTMLElement).querySelector(
			'.dropdown-toggle'
		);

		act(() => {
			if (dropdownToggle) {
				Simulate.click(dropdownToggle);
			}
		});

		expect(container).toMatchSnapshot();
	});

	it('displays custom color editor', () => {
		act(() => {
			ReactDOM.render(
				<ClayColorPicker
					colors={['008000', '00FFFF', '0000FF']}
					label="Custom Colors"
					name="colorPicker2"
					onColorsChange={() => {}}
					onValueChange={() => {}}
					spritemap="/test/path"
					title="Custom Colors"
					value="008000"
				/>,
				container
			);
		});

		const dropdownToggle = (container as HTMLElement).querySelector(
			'.dropdown-toggle'
		);

		act(() => {
			if (dropdownToggle) {
				Simulate.click(dropdownToggle);
			}
		});

		const colorEditorToggle = (container as HTMLElement).querySelector(
			'.clay-color-header button'
		);

		act(() => {
			if (colorEditorToggle) {
				Simulate.click(colorEditorToggle);
			}
		});

		expect(container).toMatchSnapshot();
	});
});
