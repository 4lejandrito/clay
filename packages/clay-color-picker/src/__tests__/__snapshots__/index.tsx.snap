// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayColorPicker displays custom color editor 1`] = `
<div>
  <div
    class="clay-color-picker"
  >
    <input
      name="colorPicker2"
      style="display: none;"
      type="text"
      value="008000"
    />
    <label>
      Custom Colors
    </label>
    <div
      class="clay-color input-group"
    >
      <div
        class="input-group-item input-group-item-shrink input-group-prepend"
      >
        <div
          class="input-group-text"
        >
          <button
            aria-label="Select a color"
            class="btn clay-color-btn dropdown-toggle"
            style="border-width: 0px; background: rgb(0, 128, 0); height: 28px; width: 28px;"
            title="008000"
          />
          <div
            class="clay-color-dropdown-menu dropdown-menu show"
          >
            <div
              class="clay-color-header"
            >
              <span
                class="component-title"
              >
                Custom Colors
              </span>
              <button
                class="close component-action"
                type="button"
              >
                <svg
                  class="lexicon-icon lexicon-icon-times"
                  role="presentation"
                >
                  <use
                    xlink:href="/test/path#times"
                  />
                </svg>
              </button>
            </div>
            <div
              class="clay-color-swatch"
            >
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="outline: auto 3px #55ADFF; border-width: 0px; background: rgb(0, 128, 0); height: 24px; width: 24px;"
                  title="008000"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(0, 255, 255); height: 24px; width: 24px;"
                  title="00FFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(0, 0, 255); height: 24px; width: 24px;"
                  title="0000FF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
            </div>
            <div
              class="clay-color-map-group"
            >
              <div
                class="clay-color-map-hsb clay-color-map"
                style="background-color: hsl(0, 100%, 50%);"
              >
                <span
                  class="clay-color-pointer clay-color-map-pointer"
                  style="background: rgb(0, 128, 0); left: -7px; top: -7px;"
                />
              </div>
              <div
                class="clay-color-map-values"
              >
                <div
                  class="form-group"
                >
                  <div
                    class="input-group"
                  >
                    <div
                      class="input-group-item"
                    >
                      <input
                        class="form-control input-group-inset input-group-inset-before"
                        type="text"
                        value="0"
                      />
                      <label
                        class="input-group-inset-item input-group-inset-item-before"
                      >
                        R
                      </label>
                    </div>
                  </div>
                </div>
                <div
                  class="form-group"
                >
                  <div
                    class="input-group"
                  >
                    <div
                      class="input-group-item"
                    >
                      <input
                        class="form-control input-group-inset input-group-inset-before"
                        type="text"
                        value="128"
                      />
                      <label
                        class="input-group-inset-item input-group-inset-item-before"
                      >
                        G
                      </label>
                    </div>
                  </div>
                </div>
                <div
                  class="form-group"
                >
                  <div
                    class="input-group"
                  >
                    <div
                      class="input-group-item"
                    >
                      <input
                        class="form-control input-group-inset input-group-inset-before"
                        type="text"
                        value="0"
                      />
                      <label
                        class="input-group-inset-item input-group-inset-item-before"
                      >
                        B
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="clay-color-range clay-color-range-hue"
            >
              <button
                class="clay-color-pointer clay-color-range-pointer"
                style="background: hsl(0, 100%, 50%); left: -7px;"
                type="button"
              />
            </div>
            <div
              class="clay-color-footer"
            >
              <div
                class="form-group"
              >
                <div
                  class="input-group"
                >
                  <div
                    class="input-group-item"
                  >
                    <input
                      class="form-control input-group-inset input-group-inset-before"
                      type="text"
                      value="008000"
                    />
                    <label
                      class="input-group-inset-item input-group-inset-item-before"
                    >
                      #
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="input-group-append input-group-item"
      >
        <input
          aria-label="Color selection is 008000"
          class="form-control input-group-inset input-group-inset-before"
          type="text"
          value="008000"
        />
        <label
          class="input-group-inset-item input-group-inset-item-before"
        >
          #
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayColorPicker displays custom colors 1`] = `
<div>
  <div
    class="clay-color-picker"
  >
    <input
      name="colorPicker2"
      style="display: none;"
      type="text"
      value="008000"
    />
    <label>
      Custom Colors
    </label>
    <div
      class="clay-color input-group"
    >
      <div
        class="input-group-item input-group-item-shrink input-group-prepend"
      >
        <div
          class="input-group-text"
        >
          <button
            aria-label="Select a color"
            class="btn clay-color-btn dropdown-toggle"
            style="border-width: 0px; background: rgb(0, 128, 0); height: 28px; width: 28px;"
            title="008000"
          />
          <div
            class="clay-color-dropdown-menu dropdown-menu show"
          >
            <div
              class="clay-color-header"
            >
              <span
                class="component-title"
              >
                Custom Colors
              </span>
              <button
                class=" component-action"
                type="button"
              >
                <svg
                  class="lexicon-icon lexicon-icon-drop"
                  role="presentation"
                >
                  <use
                    xlink:href="/test/path#drop"
                  />
                </svg>
              </button>
            </div>
            <div
              class="clay-color-swatch"
            >
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="outline: auto 3px #55ADFF; border-width: 0px; background: rgb(0, 128, 0); height: 24px; width: 24px;"
                  title="008000"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(0, 255, 255); height: 24px; width: 24px;"
                  title="00FFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(0, 0, 255); height: 24px; width: 24px;"
                  title="0000FF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="input-group-append input-group-item"
      >
        <input
          aria-label="Color selection is 008000"
          class="form-control input-group-inset input-group-inset-before"
          type="text"
          value="008000"
        />
        <label
          class="input-group-inset-item input-group-inset-item-before"
        >
          #
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayColorPicker renders 1`] = `
<div>
  <div
    class="clay-color-picker"
  >
    <input
      name="colorPicker1"
      style="display: none;"
      type="text"
      value="FFF"
    />
    <label>
      Default
    </label>
    <div
      class="clay-color input-group"
    >
      <div
        class="input-group-item input-group-item-shrink input-group-prepend"
      >
        <div
          class="input-group-text"
        >
          <button
            aria-label="Select a color"
            class="btn clay-color-btn dropdown-toggle"
            style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 28px; width: 28px;"
            title="FFF"
          />
        </div>
      </div>
      <div
        class="input-group-append input-group-item"
      >
        <input
          aria-label="Color selection is FFF"
          class="form-control input-group-inset input-group-inset-before"
          type="text"
          value="FFF"
        />
        <label
          class="input-group-inset-item input-group-inset-item-before"
        >
          #
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayColorPicker renders with menu displayed 1`] = `
<div>
  <div
    class="clay-color-picker"
  >
    <input
      name="colorPicker1"
      style="display: none;"
      type="text"
      value="FFF"
    />
    <label>
      Default
    </label>
    <div
      class="clay-color input-group"
    >
      <div
        class="input-group-item input-group-item-shrink input-group-prepend"
      >
        <div
          class="input-group-text"
        >
          <button
            aria-label="Select a color"
            class="btn clay-color-btn dropdown-toggle"
            style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 28px; width: 28px;"
            title="FFF"
          />
          <div
            class="clay-color-dropdown-menu dropdown-menu show"
          >
            <div
              class="clay-color-header"
            >
              <span
                class="component-title"
              >
                Default Colors
              </span>
            </div>
            <div
              class="clay-color-swatch"
            >
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(0, 0, 0); height: 24px; width: 24px;"
                  title="000000"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(95, 95, 95); height: 24px; width: 24px;"
                  title="5F5F5F"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(154, 154, 154); height: 24px; width: 24px;"
                  title="9A9A9A"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(203, 203, 203); height: 24px; width: 24px;"
                  title="CBCBCB"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(225, 225, 225); height: 24px; width: 24px;"
                  title="E1E1E1"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 255, 255); height: 24px; width: 24px;"
                  title="FFFFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(255, 13, 13); height: 24px; width: 24px;"
                  title="FF0D0D"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(255, 138, 28); height: 24px; width: 24px;"
                  title="FF8A1C"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(43, 166, 118); height: 24px; width: 24px;"
                  title="2BA676"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(0, 110, 248); height: 24px; width: 24px;"
                  title="006EF8"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(127, 38, 255); height: 24px; width: 24px;"
                  title="7F26FF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(255, 33, 160); height: 24px; width: 24px;"
                  title="FF21A0"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(255, 95, 95); height: 24px; width: 24px;"
                  title="FF5F5F"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(255, 180, 110); height: 24px; width: 24px;"
                  title="FFB46E"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(80, 210, 160); height: 24px; width: 24px;"
                  title="50D2A0"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(75, 155, 255); height: 24px; width: 24px;"
                  title="4B9BFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(175, 120, 255); height: 24px; width: 24px;"
                  title="AF78FF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(255, 115, 195); height: 24px; width: 24px;"
                  title="FF73C3"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(255, 177, 177); height: 24px; width: 24px;"
                  title="FFB1B1"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(255, 222, 192); height: 24px; width: 24px;"
                  title="FFDEC0"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(145, 227, 195); height: 24px; width: 24px;"
                  title="91E3C3"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(157, 200, 255); height: 24px; width: 24px;"
                  title="9DC8FF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(223, 202, 255); height: 24px; width: 24px;"
                  title="DFCAFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(255, 197, 230); height: 24px; width: 24px;"
                  title="FFC5E6"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(255, 217, 217); height: 24px; width: 24px;"
                  title="FFD9D9"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(255, 243, 232); height: 24px; width: 24px;"
                  title="FFF3E8"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(177, 235, 213); height: 24px; width: 24px;"
                  title="B1EBD5"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(197, 223, 255); height: 24px; width: 24px;"
                  title="C5DFFF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border: 1px solid #e7e7ed; background: rgb(248, 242, 255); height: 24px; width: 24px;"
                  title="F8F2FF"
                />
              </div>
              <div
                class="clay-color-swatch-item"
              >
                <button
                  class="btn clay-color-btn "
                  style="border-width: 0px; background: rgb(255, 237, 247); height: 24px; width: 24px;"
                  title="FFEDF7"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="input-group-append input-group-item"
      >
        <input
          aria-label="Color selection is FFF"
          class="form-control input-group-inset input-group-inset-before"
          type="text"
          value="FFF"
        />
        <label
          class="input-group-inset-item input-group-inset-item-before"
        >
          #
        </label>
      </div>
    </div>
  </div>
</div>
`;
